example-downloader-plugin/src/main/java/app/revanced/manager/plugin/downloader/example/DownloaderPluginImpl.kt

```diff
@@ -0,0 +1,58 @@
+package app.revanced.manager.plugin.downloader.example
+
+import android.content.pm.PackageManager
+import androidx.paging.PagingConfig
+import app.revanced.manager.plugin.downloader.DownloaderPlugin
+import app.revanced.manager.plugin.downloader.singlePagePagingSource
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import kotlinx.parcelize.Parcelize
+import java.nio.file.Files
+import java.nio.file.StandardCopyOption
+import kotlin.io.path.Path
+
+@Suppress("Unused", "MemberVisibilityCanBePrivate")
+class DownloaderPluginImpl(downloaderPluginParameters: DownloaderPlugin.Parameters) :
+    DownloaderPlugin<DownloaderPluginImpl.AppImpl> {
+    private val pm = downloaderPluginParameters.context.packageManager
+
+    private fun getPackageInfo(packageName: String) = try {
+        pm.getPackageInfo(packageName, 0)
+    } catch (_: PackageManager.NameNotFoundException) {
+        null
+    }
+
+    override val pagingConfig = PagingConfig(pageSize = 1)
+
+    override fun createPagingSource(parameters: DownloaderPlugin.SearchParameters) =
+        singlePagePagingSource {
+            val impl = withContext(Dispatchers.IO) { getPackageInfo(parameters.packageName) }?.let {
+                AppImpl(
```

Comment:
Probably PMDownloader and PMApp would be a better name for purpose of an example instead of "AppImpl"